### PR TITLE
Publish nightly docs

### PR DESCRIPTION
**Pull request by mdedetrich**
Taken from pekko-management which is published https://github.com/apache/incubator-pekko-management/blob/main/.github/workflows/publish-nightly.yml#L40-L61

### CODE REVIEW

**Comment by pjfanning** [comment]
Is it possible that this project builds with scala-2.13 by default unlike pekko-management - https://github.com/apache/incubator-pekko-connectors/blob/bd0e1a9506bb29c08f8395bbf11b23ef9c7e709a/project/Common.scala#L50

**Comment by mdedetrich** [comment]
> Is it possible that this project builds with scala-2.13 by default unlike pekko-management -
> 
> https://github.com/apache/incubator-pekko-connectors/blob/bd0e1a9506bb29c08f8395bbf11b23ef9c7e709a/project/Common.scala#L50

Aside from consistency, I don't think the default version matters that much anymore at least ever since https://github.com/apache/incubator-pekko/pull/281 and https://github.com/apache/incubator-pekko/pull/257 have been merged and propagated through the various Pekko modules.

We can make a decision on which default version of Scala should be used within our projects (I would opt for 2.13)

**Comment by pjfanning** [comment]
My point is that the scalaVersion used in the build affects which dir that the build files end up. The sbt command used in this build does not specify which scala version to use so the default is used. The default appears to be 2.13 so the target dir will be `target/scala-2.13`. The workflow changes in this PR seem to use `target/scala-2.12`.

**Comment by mdedetrich** [comment]
> My point is that the scalaVersion used in the build affects which dir that the build files end up. The sbt command used in this build does not specify which scala version to use so the default is used. The default appears to be 2.13 so the target dir will be `target/scala-2.13`. The workflow changes in this PR seem to use `target/scala-2.12`.

Ah I see, in that case I have updated the default Scala version to 2.12 since that is what we seem to use everywhere else. I also checked the github workflows to see if this has any unintended consequences and all seems to be good.

File: project/Common.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ object Common extends AutoPlugin {
     projectInfoVersion := (if (isSnapshot.value) "snapshot" else version.value),
     crossVersion := CrossVersion.binary,
     crossScalaVersions := Dependencies.ScalaVersions,
-    scalaVersion := Dependencies.Scala213,
+    scalaVersion := Dependencies.Scala212,
```
Comment (pjfanning) [inline project/Common.scala:50]: I would prefer to use Scala213 as the default. All I'm asking for is that we github workflow uses `target/scala-2.13`

Comment (pjfanning) [reply project/Common.scala:50]: main pekko repos default to scala 2.13 - it's only a few like pekko-management that default to 2.12.

Comment (mdedetrich) [reply project/Common.scala:50]: Okay sure, ill revert this and update pekko management

Comment (mdedetrich) [reply project/Common.scala:50]: Done and pushed.

**Review by pjfanning — APPROVED** [review]
lgtm

File: .github/workflows/publish-nightly.yml [hunk 2]
```diff
@@ -31,3 +31,26 @@ jobs:
...
+      - name: Build Documentation
+        run: |-
+          sbt docs/paradox unidoc
+
+      # Create directory structure upfront since rsync does not create intermediate directories otherwise
+      - name: Create nightly directory structure
+        run: |-
+          mkdir -p target/nightly-docs/docs/pekko-connectors/${{ github.ref_name }}-snapshot/
+          mv docs/target/paradox/site/main/ target/nightly-docs/docs/pekko-connectors/${{ github.ref_name }}-snapshot/docs
+          mv target/scala-2.13/unidoc target/nightly-docs/docs/pekko-connectors/${{ github.ref_name }}-snapshot/api
```
Comment (He-Pin) [inline .github/workflows/publish-nightly.yml:44]: Maybe we can define a variable in YAML for `defaultScalaVersion`?

Comment (mdedetrich) [reply .github/workflows/publish-nightly.yml:44]: The proper way to solve this issue is via an sbt plugin akin to https://github.com/lightbend/sbt-publish-rsync, there is already an issue on this https://github.com/apache/incubator-pekko/issues/104

**Review by He-Pin — APPROVED** [review]
lgtm

**Pull request merged**